NTUEELightDance/LightDance-Editor
# EDITOR-#161 Fix time shifting functionality.

### What is this PR for?
1. Fix shift time function. 
   - Shift time function needs to be connected to related api. 
   - The inputs of the function are changed from frame index to start/end time.
2. Simplify some code in /subscription.ts (see picture below)
3. Modify api code structure. 
   (Move dancerAgent, controlAgent, and positionAgent to a new file respectively, and export them in /api/index.ts)

### What type of PR is it?
Improvement

### Todos

### What is the Github issue?
Resolve https://github.com/NTUEELightDance/LightDance-Editor/issues/161

### How should this be tested?
1. click TOOLS -> TimeShift
2. choose Control/Position/Both
5. input startTime, endTime, and shiftTime(move). 
6. click OK and see if the markers changed as you want

### Screenshots (if appropriate)
1. shift time function
<img width="1440" alt="截圖 2022-03-02 下午3 47 17" src="https://user-images.githubusercontent.com/73121038/156319864-35d53457-4cd1-4e9f-b7bd-3510d1cfa33f.png">
<img width="1440" alt="截圖 2022-03-02 下午3 47 44" src="https://user-images.githubusercontent.com/73121038/156319880-aa1b0e0d-62d3-4869-baca-50080e5a85b3.png">
2. simplify code example
<img width="476" alt="截圖 2022-03-02 下午4 07 26" src="https://user-images.githubusercontent.com/73121038/156320997-734ecf52-807b-4f82-bf27-416103aa45ed.png">

### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? No
* Does this need new documentation? No
